src/adapter/src/coord/sequencer/inner/subscribe.rs

```diff
@@ -246,6 +251,8 @@ impl Coordinator {
                 return Err(AdapterError::AbsurdSubscribeBounds { as_of, up_to });
             }
         }
+        self.acquire_read_holds_auto_cleanup(ctx.session(), as_of, &bundle, true)
+            .expect("able to acquire read holds at the time that we just got from `determine_timestamp`");
```

Comment:
The read hold will be cleaned up at the end of the transaction, which for a SUBSCRIBE can be very long running, right? If a SUBSCRIBE runs for a day, we wouldn't want to hold its input frontier back to the initial as-of for the whole day too.

Would it be possible to only hold the read hold until `ship_dataflows`? After the dataflow has been installed with the controller, the controller has its own read holds so adapter can release its own.

Reply:
Fixed. Added an intermediate read hold.